Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/DayPackageSyncTool.kt

```diff
@@ -92,25 +92,25 @@ class DayPackageSyncTool @Inject constructor(
         location: LocationCode,
         forceIndexLookup: Boolean
     ): LocationDays? {
-        val cachedDays = getDownloadedCachedKeys(location, Type.LOCATION_DAY)
+        // existing or checked files -> no download needed
+        val cachedKeys = getCachedKeys(location, Type.LOCATION_DAY)
 
-        if (!forceIndexLookup && !expectNewDayPackages(cachedDays)) {
+        if (!forceIndexLookup && !expectNewDayPackages(cachedKeys)) {
             Timber.tag(TAG).d("We don't expect new day packages.")
             return null
         }
 
         val availableDays = LocationDays(location, keyServer.getDayIndex(location))
 
-        val staleDays = cachedDays.findStaleData(listOf(availableDays))
-
-        if (staleDays.isNotEmpty()) {
-            Timber.tag(TAG).d("Deleting stale days (loation=%s): %s", location, staleDays)
-            keyCache.delete(staleDays.map { it.info })
+        // remove files that are no longer available on the server
+        val staleKeys = cachedKeys.findStaleData(listOf(availableDays))
+        if (staleKeys.isNotEmpty()) {
+            Timber.tag(TAG).d("Deleting stale days (loation=%s): %s", location, staleKeys)
+            keyCache.delete(staleKeys.map { it.info })
```

Comment:
- is `checkedForSubmission` Keys part of this? , if so, we should `NOT` delete the `meta-data` as per specs , but we SHOULD delete the `files`
- I don't see where actually the `files` are deleted. Do we delete them anywhere?

Reply:
Stale Keys are those which have been removed from the server. They cannot be downloaded anymore and should also be removed from the device.
Delete will remove the meta data and the file itself.

Reply:
ok, where do we delete checkedForSubmission files?

Reply:
next PR